tests/std/tests/P0768R1_spaceship_cpos/test.cpp

```diff
@@ -0,0 +1,467 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Test the customization point objects in N4861 [cmp.alg].
+
+#include <bit>
+#include <cassert>
+#include <compare>
+#include <cstddef>
+#include <limits>
+#include <type_traits>
+#include <utility>
+
+// Intentionally not `using namespace std;` because we need to test ADL.
+using std::bit_cast, std::compare_partial_order_fallback, std::compare_strong_order_fallback,
+    std::compare_weak_order_fallback, std::declval, std::extent_v, std::is_same_v, std::numeric_limits, std::pair,
+    std::partial_ordering, std::size_t, std::strong_ordering, std::void_t, std::weak_ordering;
+
+// Define test machinery.
+struct IllFormed {};
+
+template <const auto& CPO, typename E, typename F, typename = void>
+struct CpoResultImpl {
+    using type = IllFormed;
+};
+
+template <const auto& CPO, typename E, typename F>
+struct CpoResultImpl<CPO, E, F, void_t<decltype(CPO(declval<E>(), declval<F>()))>> {

```

Comment:
Peculiar (but not wrong) that you're using `void_t` here instead of a requires-clause for the partial specialization when the test is using concepts_matrix.

Reply:
Peculiar (but not wrong) that you're recrystallizing dilithium with gamma-ray photons instead of a theta-matrix compositor, like any modern starship built by Utopia Planitia in the 2360s.

(Silly me using archaic future tech)